Customization documentation reworked (#3555)
WeblateOrg/weblate: docs/admin/customize.rst

.. _custom-module:

Creating a Python module

> ----------------------

Reviewer: ```suggestion
Creating a Python module
------------------------
```

        setup(
            name = "weblate_customization",
            version = "0.0.1",
            author = "Your name",

>             author_email = "yourname@example.com",

Author: Maybe do @orangesunny, or someone not @nijel that has written a customization that made it upstream?

            description = "Sample Custom check for Weblate.",

>             license = "AGPLv3+",

Reviewer: ```suggestion
            license = "GPLv3+",
```

To be consistent with Weblate. AGPL adds additional requirement for source code download, which Weblate doesn't provide.

Author: That doesn't have to be built in. To my knowledge network use differentiates in that it constitutes distribution, which invokes the requirement to share the code. (Per usual means.)

Author: The problem to fix here was "BSD" not being a qualified license, as it needs to be stated how many clauses it has.

            keywords = "Weblate check example",
            packages=['weblate_customization'],
        )

> 3. Create a folder for the Python module (also called `weblate_customization`)

Reviewer: ```suggestion
3. Create a folder for the Python module (also called ``weblate_customization``)
```

The quoting was wrong before as well.

> for the customization code.

Reviewer: ```suggestion
3. Create a folder for the Python module (also called ``weblate_customization``)
   for the customization code.
```

4. Within it, create a :file:`__init__.py` file to make sure Python can import the module.

> 5. This package can now be installed using :command:`pip install -e<https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>.`_

Reviewer: ```suggestion
5. This package can now be installed using :command:`pip install -e`. See more info on `editable installs <https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>.`_
```

I don't think that what you did will work well.

> 5. This package can now be installed using :command:`pip install -e`. See more info on `editable installs <https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>.`_

Reviewer: ```suggestion
5. This package can now be installed using :command:`pip install -e`. See more info on :ref:`pip:editable-installs`.
```

        ├── addons.py
        └── checks.py

> You can find an example of for custimizing Weblate at

Author: ```suggestion
You can find an example of custimizing Weblate at
```

<https://github.com/WeblateOrg/customize-example>, it covers all the topics
described below.

Changing the logo

> -------------

Reviewer: ```suggestion
Changing the logo
-----------------
```

1. Create a simple Django app containing the static files you want to overwrite
(see :ref:`custom-module`).
2. Add it to :setting:`django:INSTALLED_APPS`:

.. code-block:: python

   INSTALLED_APPS = (
      # Add your customization as first
      'weblate_customization',

      # Weblate apps are here…
   )

Branding appears in the following files:

:file:`icons/weblate.svg`
    Logo shown in the navigation bar.
:file:`logo-*.png`
    Web icons depending on screen resolution and web-browser.
:file:`favicon.ico`
    Web icon used by legacy browsers.
:file:`weblate-*.png`

>     Avatars for bots or anonymous users. Some web-browsers use these as shortcut icons.

Author: What browsers?

Reviewer: When you create a shortcut for a website on desktop (on computer) or in the launcher (on phone).

:file:`email-logo.png`
    Used in notifications e-mails.

> 3. Run :samp:`./manage.py collectstatic --noinput`, collecting static files served to

Author: ```suggestion
3. Run :samp:`./manage.py collectstatic --noinput`, to collect static files served to
```

Custom quality checks and auto fixes
------------------------------------

To install your code for :ref:`custom-autofix` or :ref:`custom-checks` and in Weblate:

1. Place the files in your Python module containing Weblate customization (see :ref:`custom-module`).

> 2. Add its fully-qualified path to the Python class in the dedicated settings

Author: Not sure about this. (Or the other instance below). They point to different ":setting"s too…

Reviewer: There is different setting for checks, fixes and addons. Maybe all these could be squashed to one chapter as they are samoe for most of the things?

Author: Saw you editing, my attempt was:

> Custom quality checks, -addons and auto-fixes
---------------------------------------------

To install your code for :ref:`custom-autofix` or :ref:`custom-checks` and in Weblate:

1. Place the files in your Python module containing the Weblate customization
   (see :ref:`custom-module`).
2. Add its fully-qualified path to the Python class in the dedicated settings
   (:setting:`WEBLATE_ADDONS`), (:setting:`CHECK_LIST` or :setting:`AUTOFIX_LIST`):

.. code-block:: python

  CHECK_LIST = (
      'weblate_customization.checks.FooCheck',
  )
  
  or
 
   WEBLATE_ADDONS = (
      'weblate_customization.addons.ExamplePreAddon',
   )

.. seealso::

   :ref:`own-checks`, :ref:`own-addon`, :ref:`addon-script`

.. _custom-addon-modules:

> (:setting:`CHECK_LIST` or :setting:`AUTOFIX_LIST`) to enable it:

Reviewer: ```suggestion
2. Add its fully-qualified path to the Python class in the dedicated settings
   (:setting:`CHECK_LIST` or :setting:`AUTOFIX_LIST`) to enable it:
```

> (:setting:`CHECK_LIST` or :setting:`AUTOFIX_LIST`) to enable it:

Reviewer: ```suggestion
2. Add its fully-qualified path to the Python class in the dedicated settings
   (:setting:`CHECK_LIST` or :setting:`AUTOFIX_LIST`) to enable it:
```

Custom addons
-------------

1. Place them in your Python module folder containing the Weblate customization.
(see :ref:`custom-module`).
2. Add its fully-qualified path to the Python class in the dedicated setting section

> (:setting:`WEBLATE_ADDONS`):

Reviewer: ```suggestion
1. Place them in your Python module folder containing the Weblate customization.
   (see :ref:`custom-module`).
2. Add its fully-qualified path to the Python class in the dedicated setting section
   (:setting:`WEBLATE_ADDONS`):
```
